### PR TITLE
Create resources using kube job

### PR DESCRIPTION
**Pull request by yitzhak12**
Create a new fixture for creating scale pods and PVCs and run IO using a Kube job. The fixture uses the `create_scale_pods` function https://github.com/red-hat-storage/ocs-ci/blob/73cfafd0f98ca485fda388f6e9cf443d87788c2f/ocs_ci/ocs/scale_lib.py#L229 for this purpose. The changes I have made should not affect other tests that use the methods I have modified. 
The goal of the pr is mainly for the managed service because it is easier to run IO using Kube job than using the old function for creating resources and running IO(But ofc we can use it also for other platforms as well).

### CODE REVIEW

File: ocs_ci/ocs/scale_lib.py [hunk 1]
```diff
@@ -267,7 +268,11 @@ def create_scale_pods(
         self.ms_name = list()
 
         # Check for expected worker count
-        expected_worker_count = get_expected_worker_count(scale_count)
+        if is_managed_service_cluster():
+            expected_worker_count = 3
+        else:
+            expected_worker_count = get_expected_worker_count(scale_count)
+
```
Comment (ramkiperiy) [inline ocs_ci/ocs/scale_lib.py:275]: This change looks ok, would like to know if we will create more than 1500+ PVCs and more than 76+ pods.

File: tests/e2e/workloads/test_create_resources_using_kube_job.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+log = logging.getLogger(__name__)
+
+
+def test_create_resources_using_kube_job(create_resources_using_kube_job):
+    config.switch_to_consumer()
+    log.info("Start creating resources...")
+    create_resources_using_kube_job()
+    ceph_health_check()
+    log.info("End of the test")
```
Comment (ramkiperiy) [inline tests/e2e/workloads/test_create_resources_using_kube_job.py:14]: PR looks good, Please execute PR validation job.

File: ocs_ci/ocs/scale_lib.py (hunk 1 above)
Comment (yitzhak12) [reply ocs_ci/ocs/scale_lib.py:275]: Currently, we don't need to use more than 1500 PVCs.

File: tests/e2e/workloads/test_create_resources_using_kube_job.py (hunk 2 above)
Comment (yitzhak12) [reply tests/e2e/workloads/test_create_resources_using_kube_job.py:14]: Okay, thanks.

**Comment by yitzhak12** [comment]
pr validation job passed on my local: http://rhsqe-repo.lab.eng.blr.redhat.com/OCS/ocs-qe-bugs/pr-5919/test2-output.log

File: tests/conftest.py [hunk 3]
```diff
@@ -5293,3 +5294,44 @@ def set_live_must_gather_images(pytestconfig):
...
+            io_runtime=io_runtime,
+            pvc_size=pvc_size,
+            max_pvc_size=max_pvc_size,
+        )
+        return kube_pod_obj_list, kube_pvc_obj_list
+
+    def finalizer():
+        log.info("Cleaning the fioscale instances")
+        for instance in instances:
+            instance.cleanup()
```
Comment (ramkiperiy) [inline tests/conftest.py:5334]: Does all the resources are getting cleaned up here?. Hope we are not leaving any left overs.

File: tests/e2e/workloads/test_create_resources_using_kube_job.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    """
+
+    @skipif_aws_i3
+    @skipif_bm
+    @skipif_lso
+    @skipif_ibm_cloud
+    @skipif_ibm_power
+    @skipif_external_mode
+    @ipi_deployment_required
+    def test_create_resources_using_kube_job(self, create_resources_using_kube_job):
```
Comment (PrasadDesala) [inline tests/e2e/workloads/test_create_resources_using_kube_job.py:39]: I dont see any managed services marker to this test, so I assume that this is a non MS test.
If that is the case, why the test has to skip on upi, external, bm, lso etc..

File: ocs_ci/ocs/scale_lib.py [hunk 5]
```diff
@@ -93,6 +94,7 @@ def create_multi_pvc_pod(
         start_io=True,
         io_runtime=None,
         pvc_size=None,
+        max_pvc_size=105,
```
[the remark this answers is not in the document]
Comment (yitzhak12) [reply ocs_ci/ocs/scale_lib.py:97]: Yes. I preserved the old value as defined here: https://github.com/red-hat-storage/ocs-ci/blob/73cfafd0f98ca485fda388f6e9cf443d87788c2f/ocs_ci/ocs/scale_lib.py#L1102.

File: ocs_ci/ocs/scale_lib.py [hunk 6]
```diff
@@ -1096,10 +1116,18 @@ def construct_pvc_creation_yaml_bulk_for_kube_job(
 
     # Construct PVC.yaml for the no_of_required_pvc count
     # append all the pvc.yaml dict to pvc_dict_list and return the list
+    if max_pvc_size <= 10:
+        raise ValueError(
+            f"The max pvc size is {max_pvc_size}, and it should be greater than 10"
+        )
```
[the remark this answers is not in the document]
Comment (yitzhak12) [reply ocs_ci/ocs/scale_lib.py:1122]: The default value is `105` as defined here: https://github.com/red-hat-storage/ocs-ci/pull/5919/files#diff-5277af9769e0fadc38a711535d8e3996b06f51b54015bf9b8e176b58e7f53ffaR1099. So in the function you mentioned or other functions that calls the `construct_pvc_creation_yaml_bulk_for_kube_jo` function will still work with the old value `105`. 
And about the first question, it seems reasonable max value according to the rand values here: https://github.com/red-hat-storage/ocs-ci/pull/5919/files#diff-5277af9769e0fadc38a711535d8e3996b06f51b54015bf9b8e176b58e7f53ffaR1127

File: tests/e2e/workloads/test_create_resources_using_kube_job.py (hunk 4 above)
Comment (yitzhak12) [reply tests/e2e/workloads/test_create_resources_using_kube_job.py:39]: I saw here https://github.com/red-hat-storage/ocs-ci/blob/73cfafd0f98ca485fda388f6e9cf443d87788c2f/ocs_ci/ocs/scale_lib.py#L271 that we use just specific platforms and deployment types. That's why I added all the skips above. Maybe we can remove some of them.

File: tests/conftest.py [hunk 7]
```diff
@@ -5293,3 +5294,62 @@ def set_live_must_gather_images(pytestconfig):
         config.REPORTING[
             "default_ocs_must_gather_latest_tag"
         ] = defaults.MUST_GATHER_UPSTREAM_TAG
+
+
+@pytest.fixture(scope="function")
+def create_resources_using_kube_job(request):
```
[the remark this answers is not in the document]
Comment (yitzhak12) [reply tests/conftest.py:5300]: Yes, we can change the name to be more accurate.

[the remark this answers is not in the document]
Comment (yitzhak12) [reply tests/conftest.py]: Done

File: tests/e2e/workloads/test_create_resources_using_kube_job.py (hunk 4 above)
Comment (yitzhak12) [reply tests/e2e/workloads/test_create_resources_using_kube_job.py]: Done

File: ocs_ci/ocs/constants.py [hunk 8]
```diff
@@ -1331,6 +1331,7 @@
 # bm dict value is based on each worker BM machine of config 40CPU and 256G/184G RAM
 # azure dict value is based on assumption similar to vmware vms min worker config of 12CPU and 64G RAM
 SCALE_WORKER_DICT = {
+    120: {"aws": 3, "vmware": 3, "bm": 2, "azure": 3, "rhv": 3},
```
Comment (ramkiperiy) [inline ocs_ci/ocs/constants.py:1334]: are we adding more workers? or we are using existing worker nodes by labeling them?
In IPI clusters new machineset will be created and and workers will be added based on this dict, with MS consumer cluster default has 3 workers, 3 infra, so are we using those 3 nodes for scale ? and labeling will be done automatic by Automation?

Comment (yitzhak12) [reply ocs_ci/ocs/constants.py:1334]: My plan is not to add worker nodes to the cluster with the value above. I was just labeling the existing ones.
With MS, I modify the code here to have three worker nodes(for any value): https://github.com/red-hat-storage/ocs-ci/pull/5919/files#diff-5277af9769e0fadc38a711535d8e3996b06f51b54015bf9b8e176b58e7f53ffaR284.

**Review by ramkiperiy — APPROVED** [review]
LGTM

**Comment by yitzhak12** [comment]
Pr validation job passed on managed service cluster: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/321/

**Review by PrasadDesala — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Review by ramkiperiy — APPROVED** [review]

**Pull request merged**